### DISCUSSION TITLE
why get this err?

### DISCUSSION THREAD

**Discussion by memeryliu** [time-rs/time#695] [Question]
error[E0282]: type annotations needed for `Box<_>`
  --> /root/.cargo/registry/src/index.crates.io-6f17d22bba15001f/time-0.3.29/src/format_description/parse/mod.rs:83:9
   |
83 |     let items = format_items
   |         ^^^^^
...
86 |     Ok(items.into())
   |              ---- type must be known at this point
   |
help: consider giving `items` an explicit type, where the placeholders `_` are specified
   |
83 |     let items: Box<_> = format_items

**Comment by jhpratt** [comment]
A simple search goes a long way: #681 #685 #689 #693

**Reply by mrdgo** [reply]
I viewed all the issues that you linked to. In none of them did I find any explanation on _why_ `Box<[Item]>` is invalid. Would you mind to elaborate?

**Reply by jhpratt** [reply]
I'm not providing support for something that has been resolved for well over a year at this point. You're free to search online for the other discussions that happened, but I'm not going back into it myself.
